plugins/database/mongodb/util.go

```diff
@@ -0,0 +1,40 @@
+package mongodb
+
+type createUserCommand struct {
+	Username string        `bson:"createUser"`
+	Password string        `bson:"pwd"`
+	Roles    []interface{} `bson:"roles"`
+}
+type mongodbRole struct {
+	Role string `json:"role" bson:"role"`
+	DB   string `json:"db"   bson:"db"`
+}
+
+type mongodbRoles []mongodbRole
+
+type mongoDBStatement struct {
+	DB    string       `json:"db"`
+	Roles mongodbRoles `json:"roles"`
+}
+
+func (roles mongodbRoles) toStandardRolesArray() []interface{} {
+	// Convert array of role documents like:
+	//
+	// [ { "role": "readWrite" }, { "role": "readWrite", "db": "test" } ]
```

Comment:
In this case the role is applied to the session's database, which will be the "db" string from the creation_statement, or "admin" if none is provided.

https://docs.mongodb.com/manual/reference/method/db.createUser/
> To specify a role that exists in a different database, specify the role with a document.